### PR TITLE
chore(aws): update ecs and ec2 packages to support task run command

### PR DESCRIPTION
**Pull request by Louise15926**
This PR includes the following changes to aws package:
1. adds a ec2 package to get subnet and security groups
2. add runtask and defaultCluster method to ecs package

Related #702 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by pkandasamy91** [comment]
Can you make the title more descriptive (this will be the commit title, so it would be helpful to users to be able to understand what is/has changed)

File: internal/pkg/aws/ec2/ec2.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	}
+)
+
+type api interface {
+	DescribeSubnets(*ec2.DescribeSubnetsInput) (*ec2.DescribeSubnetsOutput, error)
+	DescribeSecurityGroups(*ec2.DescribeSecurityGroupsInput) (*ec2.DescribeSecurityGroupsOutput, error)
+}
+
+// Filter contains the name and values of a filter.
+type Filter struct {
```
Comment (kohidave) [inline internal/pkg/aws/ec2/ec2.go:41]: Since this is the input to a function - that's used out of the EC2 context - I'm not sure what I should put as Name and Value. 

Perhaps you can add comments to them

```golang
// Name is the name of the subnet bla bla bla
Name string
// Values are the bla bla bla
Values []string
```

File: internal/pkg/aws/ecs/ecs.go [hunk 2]
```diff
@@ -168,6 +181,58 @@ func (e *ECS) ServiceTasks(clusterName, serviceName string) ([]*Task, error) {
 	return tasks, nil
 }
 
+// DefaultCluster returns the default clusters in the account and region.
+func (e *ECS) DefaultCluster() (string, error) {
+	resp, err := e.client.DescribeClusters(&ecs.DescribeClustersInput{})
```
Comment (kohidave) [inline internal/pkg/aws/ecs/ecs.go:186]: I think if we want this to actually be the default cluster, we have to include `default` in the list of cluster names passed in.

Otherwise we just get *all* the clusters and we're just grabbing a random one. 
https://docs.aws.amazon.com/sdk-for-go/api/service/ecs/#DescribeClustersInput

File: internal/pkg/aws/ecs/ecs.go [hunk 3]
```diff
@@ -168,6 +181,58 @@ func (e *ECS) ServiceTasks(clusterName, serviceName string) ([]*Task, error) {
...
+	if err != nil {
+		return "", fmt.Errorf("get default cluster: %w", err)
+	}
+
+	if len(resp.Clusters) == 0 {
+		return "", errors.New("no default cluster is found")
```
Comment (kohidave) [inline internal/pkg/aws/ecs/ecs.go:192]: Can you make this error a const at the top? This will let us compare errors in the caller.

Like I can imagine us doing something like:

```golang

 cluster, err := ecs.DefaultCluster()
 if err == ecs.NoDefaultClusterErr {
    cluster = ecs.CreateDefaultCluster()
  }
```

File: internal/pkg/aws/ecs/ecs.go [hunk 4]
```diff
@@ -168,6 +181,58 @@ func (e *ECS) ServiceTasks(clusterName, serviceName string) ([]*Task, error) {
...
+	if err != nil {
+		return []string{}, fmt.Errorf("run task(s) %s: %w", input.TaskFamilyName, err)
+	}
+
+	taskArns := make([]string, len(resp.Tasks))
+	for idx, task := range resp.Tasks {
+		taskArns[idx] = aws.StringValue(task.TaskArn)
+	}
+
+	if err := e.client.WaitUntilTasksRunning(&ecs.DescribeTasksInput{
```
Comment (kohidave) [inline internal/pkg/aws/ecs/ecs.go:226]: Are there any specific errors that we should handle differently? I can imagine that if we fail to spin up a task maybe we can show a nicer error that our go errors.

"Failed to launch task with - essential container bla-bla-bla failed to start up" or something like that?

File: internal/pkg/aws/ecs/ecs.go (hunk 2 above)
Comment (Louise15926) [reply internal/pkg/aws/ecs/ecs.go:186]: My understanding from the doc is that if we don't specify a cluster, the default cluster is assumed. I also tested it - if I provide an empty `DescribeClustersInput`, the client will actually return the one default cluster (named "default"), instead of all clusters.

With that being said though, it is indeed easier to follow if we explicitly specify `default` in the input. My concern was that if they change the way in which they identify the default cluster (right now it is simply the cluster named `default`, but what if it is changed to some other names in the future), we would have to make the changes accordingly.

File: internal/pkg/aws/ecs/ecs.go [hunk 5]
```diff
@@ -168,6 +181,58 @@ func (e *ECS) ServiceTasks(clusterName, serviceName string) ([]*Task, error) {
...
+		NetworkConfiguration: &ecs.NetworkConfiguration{
+			AwsvpcConfiguration: &ecs.AwsVpcConfiguration{
+				AssignPublicIp: aws.String(ecs.AssignPublicIpEnabled),
+				Subnets:        aws.StringSlice(input.Subnets),
+				SecurityGroups: aws.StringSlice(input.SecurityGroups),
+			},
+		},
+	})
+	if err != nil {
+		return []string{}, fmt.Errorf("run task(s) %s: %w", input.TaskFamilyName, err)
```
[the remark this answers is not in the document]
Comment (Louise15926) [reply internal/pkg/aws/ecs/ecs.go:218]: I must be in a state of unconsciousness when I did this. Changed it to `nil`!

File: internal/pkg/aws/ecs/ecs.go (hunk 2 above)
Comment (kohidave) [reply internal/pkg/aws/ecs/ecs.go:186]: Ah! Such a good catch - thank you!!! You're right :)

> If you do not specify a cluster, the default cluster is assumed.

File: internal/pkg/aws/ecs/ecs.go (hunk 4 above)
Comment (Louise15926) [reply internal/pkg/aws/ecs/ecs.go:226]: From the doc I didn't find super helpful errors that are either common to all actions or are specific to this waiter https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_DescribeTasks.html... 

Do you think it might be helpful to direct the user to see the management console? In ECS > task > the specific task, it shows reasons why the task's been stopped under the Status field, which I think is not returned from the waiter API call.

File: internal/pkg/aws/ecs/errors.go [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package ecs
```
Comment (efekarakus) [inline internal/pkg/aws/ecs/errors.go:1]: copyright header plz 🙏

File: internal/pkg/aws/ecs/errors.go [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+
+import "fmt"
+
+// ErrNoDefaultCluster occurs when the default cluster is not found.
+type ErrNoDefaultCluster struct {}
```
Comment (efekarakus) [inline internal/pkg/aws/ecs/errors.go:6]: can we have: `var ErrNoDefaultCluster = errors.New("default cluster does not exist")`

File: internal/pkg/aws/ecs/errors.go (hunk 6 above)
Comment (Louise15926) [reply internal/pkg/aws/ecs/errors.go:1]: 😭😭😭😭😭😭 right i'm sorry

File: internal/pkg/aws/ec2/ec2.go [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+// New returns a EC2 configured against the input session.
+func New(s *session.Session) *EC2 {
+	return &EC2{
+		client: ec2.New(s),
+	}
+}
+
+// GetSubnetIDs finds the subnet IDs with optional filters
```
Comment (efekarakus) [inline internal/pkg/aws/ec2/ec2.go:60]: nit: end comments with period :P 
https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences

File: internal/pkg/aws/ecs/ecs.go [hunk 9]
```diff
@@ -168,6 +181,58 @@ func (e *ECS) ServiceTasks(clusterName, serviceName string) ([]*Task, error) {
 	return tasks, nil
 }
 
+// DefaultCluster returns the default clusters in the account and region.
```
Comment (efekarakus) [inline internal/pkg/aws/ecs/ecs.go:184]: nit: can we mention that it returns the ARN in the comment
